### ISSUE TITLE
Please provide independent variables for basis.

### ISSUE THREAD

**Issue by baggepinnen** [SciML/DataDrivenDiffEq.jl#146]
When I run this line
https://github.com/SciML/DataDrivenDiffEq.jl/blob/ebf63074cfa3333f36b0335308e6230556f030a5/src/koopman/extended_dmd.jl#L17
I get
```
julia> basis = Basis(h, u, parameters = p, iv = t)
ERROR: AssertionError: Please provide independent variables for basis.
Stacktrace:
 [1] Basis(::Array{Operation,1}, ::Array{Operation,1}; parameters::Array{Operation,0}, iv::Operation, eval_expression::Bool) at /home/fredrikb/.julia/packages/DataDrivenDiffEq/AhQqO/src/basis.jl:70
 [2] top-level scope at none:1
```
I was under the impression that `iv=t` is how you provide the independent variable for the basis?

The following docstring example also throws an error
https://github.com/SciML/DataDrivenDiffEq.jl/blob/ebf63074cfa3333f36b0335308e6230556f030a5/src/basis.jl#L54

**Comment by AlCap23** [comment]
Thanks for pointing that out.  I will try to fix this ASAP.

This should work and is related to simpler times, where I did not use explicit time dependencies / independent variables.

**Comment by AlCap23** [comment]
I tried different approaches, but end up with an error like

```julia
ERROR: LoadError: MethodError: objects of type Float64 are not callable
Stacktrace:
 [1] macro expansion at /home/alcapitan/.julia/packages/ModelingToolkit/kfLoQ/src/build_function.jl:125 [inlined]
 [2] macro expansion at /home/alcapitan/.julia/packages/GeneralizedGenerated/wp5nX/src/ngg/runtime_fns.jl:72 [inlined]
 [3] (::ggfunc)(::Array{Float64,1}, ::Array{Float64,1}) at /home/alcapitan/.julia/packages/GeneralizedGenerated/wp5nX/src/ngg/runtime_fns.jl:64
 [4] top-level scope at /home/alcapitan/Documents/code/DataDrivenDiffEq.jl/examples/Basis_Creation.jl:15
 [5] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1088
 [6] include_string(::Module, ::String, ::String) at ./loading.jl:1096
in expression starting at /home/alcapitan/Documents/code/DataDrivenDiffEq.jl/examples/Basis_Creation.jl:15
```

MWE would be ( after commenting out the `is_independent` function )

```julia
using DataDrivenDiffEq
using ModelingToolkit

# First, we define a set of variables and parameters
@parameters t w[1:2]
@variables u[1:3](t)

# Now, the equations which form our basis
h = [u[1]; u[2]; cos(w[1]*u[2]+w[2]*u[3])]

# Then, we simply create a basis
b= Basis(h, u, parameters = w, iv = t)

# Basis are callable with out of place
x = b([1;2;3])
```
This also happens for an `OptimizationSystem`. 


@ChrisRackauckas Any ideas regarding this ? I am out of my MTK skillset here. Maybe I will move this into MTK.

**Comment by ChrisRackauckas** [comment]
Yes, get an issue open on MTK

**Comment by AlCap23** [comment]
@baggepinnen With the next release of DataDriven and MTK, this is possible now. I just adapt the tests a little and merge the corresponding PR.

**Comment by AlCap23** [comment]
Closed via #174
